controllers/machinehealthcheck_controller_test.go

```diff
@@ -206,6 +466,50 @@ func cleanupTestMachineHealthChecks(ctx context.Context, c client.Client) error
 	return nil
 }
 
+func cleanupTestMachines(ctx context.Context, c client.Client) error {
+	machineList := &clusterv1.MachineList{}
+	err := c.List(ctx, machineList)
+	if err != nil {
+		return err
+	}
+	for _, machine := range machineList.Items {
+		m := machine
+		err = c.Delete(ctx, &m)
+		if err != nil && apierrors.IsNotFound(err) {
+			return nil
+		} else if err != nil {
+			return err
+		}
+		Eventually(func() error {
+			err := c.Get(ctx, util.ObjectKey(&m), &m)
+			if err != nil && apierrors.IsNotFound(err) {
+				return nil
+			} else if err != nil {
+				return err
+			}
```

Comment:
can be:
```
if err := c.Get(ctx, util.ObjectKey(&m), &m); err != nil {
	if apierrors.IsNotFound(err) {
		return nil
	}
	return err
}
```

Reply:
@neolit123 Thanks for the feedback, I did it slightly differently to your suggestion, WDYT?
```
if err := c.Get(ctx, util.ObjectKey(&m), &m); err != nil && apierrors.IsNotFound(err) {
    return nil
} else if err != nil {
    return err
}
```

Reply:
i prefer my method, but i can also now see the original was already used in number of places in this repo. no strong preferences on my side.